modules/sfp_zonecruncher.py

```diff
@@ -0,0 +1,186 @@
+            )
+            return None
+
+        return info
+
+    def query_subdomains(self, domain):
+        info = self.request("/subdomain", {"q": domain})
+        return info
+
+    def query_hostname(self, hostname):
+        info = self.request("/hostname", {"q": hostname})
+        return info
+
+    def query_email_domain(self, email_domain):
+        info = self.request("/email_domain", {"q": email_domain})
+        return info
+
+    def query_email_address(self, email_address):
+        info = self.request("/email_address", {"q": email_address})
+        return info
+
+    def generate_subdomains_events(self, data, pevent):
+        if isinstance(data, dict):
+            results = data.get("results", [])
+            if isinstance(results, list):
+                for r in results:
+                    qname = r.get("qname")
+                    if not isinstance(qname, str):
+                        continue
+                    self.emit("INTERNET_NAME", qname, pevent)
+
+    def generate_hostname_events(self, data, pevent):
+        hostnames = set()
+        if isinstance(data, dict):
+            results = data.get("results")
+            if isinstance("results", list):
+                for r in results:
+                    qname = r.get("qname")
+                    if isinstance("qname", str):
+                        hostnames.add(qname)
+        for hostname in hostnames:
+            self.emit("INTERNET_NAME", hostname, pevent)
+
+    def generate_email_events(self, data, pevent):
+        if isinstance(data, dict):
+            results = data.get("results")
+            if isinstance(results, list):
+                for r in results:
+                    domain = r.get("d")
+                    if isinstance(domain, str):
+                        self.emit("AFFILIATE_DOMAIN_NAME", domain, pevent)
+
+    def generate_email_domain_events(self, data, pevent):
+        if isinstance(data, dict):
+            results = data.get("results")
+            if isinstance(results, list):
+                for r in results:
+                    domain = r.get("d")
+                    if isinstance(domain, str):
+                        self.emit("AFFILIATE_DOMAIN_NAME", domain, pevent)
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error("You enabled sfp_template but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        if (eventName, eventData) in self.results:
+            self.sf.debug(f"Skipping {eventName}:{eventData}, already checked.")
+            return None
+        self.results[eventName, eventData] = True
```

Comment:
Please make the key a string instead of a tuple, as there is likely code elsewhere (or HX) that relies on this being a string for now.